### PR TITLE
docs(typos): Run codespell to fix typos

### PR DESCRIPTION
**Pull request by CharlieTLe**
There were a lot of typos through the repository, so I ran [codespell][], a tool for automatically fixing typos, to fix them.

```console
make codespell
```

There's already a tool called [misspell][] that's supposed to take care of this, but misspell hasn't been updated for 6 years, and it doesn't seem to be catching any of the typos that codespell can.

To make this more automatic, perhaps we can set up a [workflow][] to run codespell as well?

cc: @alolita 

[codespell]: https://github.com/codespell-project/codespell
[misspell]: https://github.com/client9/misspell
[workflow]: https://github.com/codespell-project/actions-codespell

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -580,7 +580,7 @@ of the `"go.opentelemetry.io/otel/bridge/opentracing".BridgeTracer` type. (#2911
 - The `LabelFilterSelector` interface from `go.opentelemetry.io/otel/sdk/metric/processor/reducer` is renamed to `AttributeFilterSelector`.
   The method included in the renamed interface also changed from `LabelFilterFor` to `AttributeFilterFor`. (#2790)
 - The `Metadata.Labels` method from the `go.opentelemetry.io/otel/sdk/metric/export` package is renamed to `Metadata.Attributes`.
-  Consequentially, the `Record` type from the same package also has had the embedded method renamed. (#2790)
+  Consequently, the `Record` type from the same package also has had the embedded method renamed. (#2790)
```
Comment (dmathieu) [inline CHANGELOG.md:583]: I'm, not sure Consequentially is wrong here (nor would it be everywhere else)

File: Makefile [hunk 2]
```diff
@@ -176,6 +176,10 @@ vanity-import-fix: | $(PORTO)
 misspell: | $(MISSPELL)
 	@$(MISSPELL) -w $(ALL_DOCS)
 
+.PHONY: codespell
+codespell:
+	codespell -S "./.git/*,go.mod,go.sum,./semconv/*" -L ot,fo,te,collison -i2 -w
```
Comment (dmathieu) [inline Makefile:181]: This is nice, but needs to run manually. Should we add a GH Action?

File: CHANGELOG.md (hunk 1 above)
Comment (CharlieTLe) [reply CHANGELOG.md:583]: I can revert these :)

**Review by pellared — COMMENTED** [review]
Can we move the automation (`Makefile` and GitHub Actions) to a separate PR?

**Comment by CharlieTLe** [comment]
> Can we move the automation (`Makefile` and GitHub Actions) to a separate PR?

Yes, I'll remove it from this PR for now and file another PR afterwards. Thanks @pellared

**Review by pellared — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Comment by CharlieTLe** [comment]
Thanks for approving @pellared @dmathieu! Since I don't have write access, could someone help merge the PR?

**Pull request merged**